core/src/commonMain/kotlin/com/powersync/db/internal/ConnectionPool.kt

```diff
@@ -56,6 +56,28 @@ internal class ConnectionPool(
         }
     }
 
+    suspend fun <R> withAllConnections(action: suspend (connections: List<TransactorDriver>) -> R): R {
+        val obtainedConnections = mutableListOf<Pair<TransactorDriver, CompletableDeferred<Unit>>>()
+
+        try {
+            // Try and get all the connections
```

Comment:
Maybe expand this comment to mention that it gets all connections because we're holding on to all of them simultaneously, so there's no way we'd accidentally receive a connection twice.

(or maybe don't if you think that's obvious, it took me a short while to grasp that this is sound)

Reply:
That's a fair point. I've expanded the comment :)